AirCasting/Map/GoogleMap/GoogleMapButler.swift

```diff
@@ -0,0 +1,154 @@
+// Created by Lunar on 27/07/2022.
+//
+import Foundation
+import SwiftUI
+import GoogleMaps
+import Resolver
+
+class GoogleMapButler: ObservableObject {
+    @Binding var notes: [MapNote]
+    
+    private let coreMap: GoogleMapCore
+    private let showMyLocationButton: Bool
+    
+    private var pathPoints: [PathPoint]
+    private var isMyLocationEnabled: Bool
+    private var liveModeOn: Bool
+    private var isSessionFixed: Bool
+    private(set) var threshold: SensorThreshold?
+    var notesNumber: Int { notes.count }
+    
+    init(pathPoints: [PathPoint],
+         showMyLocationButton: Bool,
+         isMyLocationEnabled: Bool = false,
+         isSessionActive: Bool = false,
+         isSessionFixed: Bool = false,
+         threshold: SensorThreshold? = nil,
+         notes: Binding<[MapNote]>) {
+        
+        self.pathPoints = pathPoints
+        self.showMyLocationButton = showMyLocationButton
+        self.isMyLocationEnabled = isMyLocationEnabled
+        self.threshold = threshold
+        self.liveModeOn = isSessionActive
+        self.isSessionFixed = isSessionFixed
+        self._notes = notes
+        self.coreMap = GoogleMapCoreDefault()
+    }
+    
+    var cameraUpdate: GMSCameraUpdate {
+        let initialBounds = GMSCoordinateBounds()
+        guard !pathPoints.isEmpty else {
+            let location = coreMap.getLocationForEmptyPathPoints()
+            return GMSCameraUpdate.setTarget(location)
+        }
+        
+        guard liveModeOn else {
+            let pathPointsBoundingBox = pathPoints.reduce(initialBounds) { bounds, point in
+                bounds.includingCoordinate(point.location)
+            }
+            return GMSCameraUpdate.fit(pathPointsBoundingBox, withPadding: 1.0)
+        }
+        let pathPointsBoundingBox = initialBounds.includingCoordinate(pathPoints.last!.location)
+        return GMSCameraUpdate.fit(pathPointsBoundingBox, withPadding: 1.0)
+    }
+    
+    func initialMakeUIVIew(context: UIViewRepresentableContext<GoogleMapView>) -> GMSMapView {
+        let mapView = coreMap.initialCommonMakeUIView()
+        mapView.settings.myLocationButton = showMyLocationButton
+        placeNotes(mapView, context: context)
+        drawPolyline(mapView, context: context)
+        mapView.isMyLocationEnabled = isMyLocationEnabled
+        return mapView
+    }
+    
+    func applyStylying(to mapView: GMSMapView) { coreMap.apllyStylying(to: mapView) }
```

Comment:
this is a second `applyStyling` I see, but this one is not related to the `CoreGoogleMap` protocol. If's kind of confusing what does what

Reply:
The idea was, that GoogleMapView can use only Butler methods so I was in need to repeat somehow this JSON styling from GoogleMapCore